admin/election_controller.go

```diff
@@ -0,0 +1,188 @@
+func NewElectionController(store model.ElectionStore) *ElectionController {
+	return &ElectionController{store: store}
+}
+
+// Create creates a new election with it's users
+func (c *ElectionController) Create(msg *bearerstdapi.BearerStandardAPIdata, ctx *httprouter.HTTPContext) error {
+	newElection := model.Election{}
+	if err := json.Unmarshal(msg.Data, &newElection); err != nil {
+		return err
+	}
+
+	election, err := c.store.CreateElection(&newElection)
+	if err != nil {
+		return err
+	}
+
+	electionBearerToken, err := GenerateAdminToken(election.ID)
+	if err != nil {
+		return err
+	}
+
+	response := ElectionWithTokenResponse{
+		AdminToken: electionBearerToken,
+		Election:   *election,
+	}
+
+	return ctx.Send(new(ApiResponse).Set(response).MustMarshall(), bearerstdapi.HTTPstatusCodeOK)
+}
+
+// Election returns the election data
+func (c *ElectionController) Election(msg *bearerstdapi.BearerStandardAPIdata, ctx *httprouter.HTTPContext) error {
+	var electionID types.HexBytes
+	if err := electionID.FromString(ctx.URLParam("electionId")); err != nil {
+		return err
+	}
+	election, err := c.store.Election(electionID)
+	if err != nil {
+		return err
+	}
+
+	return ctx.Send(new(ApiResponse).Set(election).MustMarshall(), bearerstdapi.HTTPstatusCodeOK)
+}
+
+// Delete deletes an election
+func (c *ElectionController) Delete(msg *bearerstdapi.BearerStandardAPIdata, ctx *httprouter.HTTPContext) error {
+	var electionID types.HexBytes
+	if err := electionID.FromString(ctx.URLParam("electionId")); err != nil {
+		return err
+	}
+
+	valid, err := ValidateAdminToken(electionID, msg.AuthToken)
+	if !valid || err != nil {
+		return ctx.Send(
+			new(ApiResponse).SetError(CodeErrInvalidAuth, ReasonErrInvalidAuth).MustMarshall(),
+			bearerstdapi.HTTPstatusCodeErr,
+		)
+	}
+
+	if err := c.store.DeleteElection(electionID); err != nil {
+		return err
+	}
+
+	return ctx.Send(new(ApiResponse).Set(nil).MustMarshall(), bearerstdapi.HTTPstatusCodeOK)
+}
+
+// List returns the list of elections
+func (c *ElectionController) List(msg *bearerstdapi.BearerStandardAPIdata, ctx *httprouter.HTTPContext) error {
+	elections, err := c.store.ListElection()
+	if err != nil {
+		return err
+	}
+
+	return ctx.Send(new(ApiResponse).Set(elections).MustMarshall(), bearerstdapi.HTTPstatusCodeOK)
+}
+
+type AdminTokenRequest struct {
+	Signature string `json:"signature"`
+	Data      string `json:"data"`
+}
+
+// AdminToken returns the admin token for an election
+func (c *ElectionController) AdminToken(msg *bearerstdapi.BearerStandardAPIdata, ctx *httprouter.HTTPContext) error {
+	var electionID types.HexBytes
+	if err := electionID.FromString(ctx.URLParam("electionId")); err != nil {
+		return err
+	}
+
+	adminTokenRequest := AdminTokenRequest{}
+	if err := json.Unmarshal(msg.Data, &adminTokenRequest); err != nil {
+		return err
+	}
+
+	// Remove the "0x" prefix from the signature if present
+	signature := strings.TrimPrefix(adminTokenRequest.Signature, "0x")
```

Comment:
All this signature check logic can be replaced by the crypto/ethereum package of vocdoni-node